### DISCUSSION TITLE
The websocket inferface doesn't generate the doc, but the other controller is ok? why

### DISCUSSION THREAD

**Discussion by skydream-xu** [TongchengOpenSource/smart-doc#898] [Q&A] [answered]
The websocket inferface doesn't generate the doc, but the other controller is ok? why

@Slf4j
@ServerEndpoint("/documents/did/{did}/bid/{bid}")
@RestController
public class DocumentWebSocketController {

    @OnMessage
    public void receive(String message, Session session, @PathParam("did") int documentId,
                        @PathParam("bid") String branchBid) {

**Comment by linwumingshi** [comment]
@skydream-xu 


In WebSocket programming, it's common to handle URL parameters within the `@OnOpen` method, where the WebSocket session is established, rather than in the `@OnMessage` method. Here’s an example of how you might do this:

```java
@ServerEndpoint("/documents/did/{did}/bid/{bid}")
@RestController
public class DocumentWebSocketController {

    private int documentId;
    private String branchBid;


    /**
     * Handles the opening of a WebSocket connection.
     *
     * @param session   the WebSocket session
     * @param documentId the ID of the document
     * @param branchBid the ID of the branch
     */
    @OnOpen
    public void onOpen(Session session, @PathParam("did") int documentId,
                       @PathParam("bid") String branchBid) {
        this.documentId = documentId;
        this.branchBid = branchBid;
    }

    @OnMessage
    public void receive(String message, Session session) {
        // You can now use documentId and branchBid here
        // Example usage:
        System.out.println("Received message: " + message + " for Document ID: " + documentId + " and Branch ID: " + branchBid);
    }

    @OnClose
    public void onClose(Session session) {
        // Handle close event
    }
}
```

By defining the parameters in the `@OnOpen` method, you can store them as instance variables and use them across the WebSocket session, including in the `@OnMessage` method. This approach helps keep the `@OnMessage` method focused on processing messages rather than extracting path parameters.



You can refer to this example: 
[ @OnOpen](https://github.com/smart-doc-group/smart-doc-example-cn/blob/36dc4aa009d99f02ed6374f327306199e7c97695/src/main/java/com/power/doc/websocket/WebSocketServerEndpointHasUrlPathVariable.java#L32)

**Comment by shalousun** [comment, marked as answer]
@skydream-xu First, upgrade to the latest version 3.0.7. Then make sure you are using the correct commands. The WebSocket and Restful interfaces currently use different execution commands. To generate the WebSocket interface, use commands like `mvn smart-doc:websocket-markdown`, `mvn smart-doc:websocket-html`, etc.

Additionally, if you used `packageFilters`, please check if it's because `packageFilters` filtered them out. If you can't find the reason after checking everything, you can debug to find the cause. For debugging, you can refer to the official guide at https://smart-doc-group.github.io/guide/plugins/maven#plug-in-debugging.

**Reply by skydream-xu** [reply]
oh， I got it . I had not use the right command that I treat the websocket the same as the rest in my mind. Thank you !
By the way , in feature do you have the plan the generate all the api(rest, websocket, probuf) in one command.

**Reply by shalousun** [reply]
The community plans to merge multiple protocols into the documentation generation in the future. However, since the initial design was primarily based on Restful, this merger will involve significant changes, including adjustments to the integration protocol with Tornado. Therefore, this will require some time.
